src/niMidi.cpp

```diff
@@ -234,21 +311,58 @@ class NiMidiSurface: public BaseSurface {
 			this->_sendSysex(CMD_TRACK_NAME, 0, numInBank, name);
 			// todo: level meters, volume, pan
 		}
-		// todo: navigate tracks, navigate banks
+		// todo: navigate tracks, navigate banks. NOTE: probably not here
+	}
+
+	void ClearSelected() {
+		// Clear all selected tracks. Copyright (c) 2010 and later Tim Payne (SWS)
+		int iSel = 0;
+		for (int i = 0; i <= GetNumTracks(); i++) // really ALL tracks, hence no use of CSurf_NumTracks
+			GetSetMediaTrackInfo(CSurf_TrackFromID(i, false), "I_SELECTED", &iSel);
+	}
+
+	void _onTrackSelect(unsigned char numInBank) {
+		int id = this->_bankStart + numInBank;
+		if (id > GetNumTracks()) {
```

Comment:
I think we need to be consistent here: if we're going to retrieve the track using CSurf_TrackFromID, we should get the number of tracks with CSurf_NumTracks. Otherwise, we're using a different numbering system.

Reply:
You are right, good catch. Right now there is no problem because CSurf_NumTracks(false) uses TCP view counting. But if we ever (see above) allowed MCP view, then it would be inconsistent.